packages/discovery/src/flatten/ParsedFilesManager.ts

```diff
@@ -0,0 +1,371 @@
+
+      return {
+        ast: declaration,
+        name: declaration.name,
+        type: declaration.kind as ContractType,
+        inheritsFrom: declaration.baseContracts.map(
+          (bc) => bc.baseName.namePath,
+        ),
+        referencedContracts: [],
+        byteRange: {
+          start: c.range[0],
+          end: c.range[1],
+        },
+      }
+    })
+  }
+
+  resolveFileImports(
+    file: ParsedFile,
+    remappings: Remapping[],
+    alreadyImportedObjects: Map<string, string[]>,
+  ): ImportDirective[] {
+    const importDirectives = file.rootASTNode.children.filter(
+      (n) => n.type === 'ImportDirective',
+    )
+
+    return importDirectives.flatMap((i) => {
+      assert(
+        i.type === 'ImportDirective' && i.range !== undefined,
+        'Invalid import directive',
+      )
+
+      const remappedPath = resolveRemappings(i.path, remappings)
+      const importedFile = this.resolveImportPath(file, remappedPath)
+
+      const alreadyImported =
+        alreadyImportedObjects.get(importedFile.path) ?? []
+      assert(
+        alreadyImported.length <= importedFile.contractDeclarations.length,
+        'Already imported more than there are contracts in the file',
+      )
+      const gotEverything =
+        alreadyImported.length === importedFile.contractDeclarations.length
+      if (gotEverything) {
+        return []
+      }
+
+      const result = []
+      const importEverything = i.symbolAliases === null
+      if (importEverything) {
+        for (const contract of importedFile.contractDeclarations) {
+          const object = {
+            absolutePath: importedFile.path,
+            originalName: contract.name,
+            importedName: contract.name,
+          }
+
+          if (!alreadyImported.includes(object.originalName)) {
+            result.push(object)
+          }
+        }
+
+        alreadyImportedObjects.set(
+          importedFile.path,
+          importedFile.contractDeclarations.map((c) => c.name),
+        )
+
+        const recursiveResult = this.resolveFileImports(
+          importedFile,
+          remappings,
+          alreadyImportedObjects,
+        )
+        return result.concat(recursiveResult)
+      }
+
+      assert(i.symbolAliases !== null, 'Invalid import directive')
+      for (const alias of i.symbolAliases) {
+        const object = {
+          absolutePath: importedFile.path,
+          originalName: alias[0],
+          importedName: alias[1] ?? alias[0],
+        }
+
+        if (!alreadyImported.includes(object.originalName)) {
+          alreadyImported.push(object.originalName)
+          result.push(object)
+        }
+      }
+
+      alreadyImportedObjects.set(importedFile.path, alreadyImported)
+
+      return result
+    })
+  }
+
+  resolveReferencedLibraries(
+    file: ParsedFile,
+    c: ContractDefinition,
+  ): string[] {
+    const identifiers = new Set(
+      c.subNodes.flatMap((n) => getASTIdentifiers(n)).map(extractNamespace),
```

Comment:
This is the only place we use `getASTIdentifiers()`. Do we really need all the `getASTIdentifiers()` complexity to pull referenced libraries?